ReactAndroid/src/main/java/com/facebook/react/devsupport/MultipartStreamReader.java

```diff
@@ -0,0 +1,127 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+package com.facebook.react.devsupport;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import okio.Buffer;
+import okio.BufferedSource;
+import okio.ByteString;
+
+/**
+ * Utility class to parse the body of a response of type multipart/mixed.
+ */
+public class MultipartStreamReader {
```

Comment:
Sadly no, this class can only be used to create a multipart request and not to parse a response. I've seen in an issue that this kind of response is not common enough to be implemented in okhttp and they recommended implementing it manually. iOS also implements this response parsing from scratch so this is largely based on it.